src/main/java/fi/fmi/avi/archiver/file/FileParser.java

```diff
@@ -49,19 +52,44 @@ public class FileParser {
     private static final String BULLETIN_ELEMENT_NAME = "MeteorologicalBulletin";
     private static final ConversionHints CONVERSION_HINTS = ConversionHints.ALLOW_ERRORS;
     private static final MessageType UNKNOWN_MESSAGE_TYPE = new MessageType("UNKNOWN");
+    private static final String IWXXM_NS_PREFIX = "icao.int/iwxxm/";
+    private static final String GML_NS_PREFIX = "opengis.net/gml/";
+    private static final String XPATH_OBSERVATION_TIME = String.format("""
+            normalize-space((
+              /*[
+                contains(namespace-uri(),'%1$s')
+                and (local-name()='METAR' or local-name()='SPECI')
+              ]
+              /*[
+                contains(namespace-uri(),'%1$s')
+                and local-name()='observationTime'
+              ]
+              /*[
+                contains(namespace-uri(),'%2$s')
+                and local-name()='TimeInstant'
+              ]
+              /*[
+                contains(namespace-uri(),'%2$s')
+                and local-name()='timePosition'
+              ]
+            )[1])
+            """, IWXXM_NS_PREFIX, GML_NS_PREFIX);
 
     private final AviMessageConverter aviMessageConverter;
     private final DocumentBuilderFactory documentBuilderFactory;
+    private final XPathFactory xPathFactory;
+    private final XPathExpression observationTimeExpression;
```

Comment:
This is something that we have missed earlier. I think thread-safety constraints inherited from XPath APIs to should be documented in the class javadoc.

In practice this is not an issue, as long as we're executing a single instance within a single dedicated thread. But just in case this ever changes, there's a small chance a javadoc comment gets noticed.

Additionally, ´XPathExpression` instance is always recreated in `getCollectIdentifier()`. This could be a good chance to harmonize the usage within the class. Reuse seems a good approach to me, as long as there's no obstacles for it.

Reply:
I added a minimal class javadoc and harmonized the usage.